src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -25,6 +25,57 @@
 from ..exceptions import ScopeNotValid, PermissionNotValid
 from ..services.auth import AuthManager, JwtManager
 
+def EndpointDoc(descr="", group="", param={}, respons={}):
+    def _split_param(params):
+        param_list = []
+        for name, props in params.items():
+            if isinstance(props, tuple):
+                param_list.append({
+                    'name': name,
+                    'type': props[0],
+                    'description': props[1]
+                })
+            elif isinstance(props, dict):
+                param_list.append({
+                    'name': name,
+                    'type': dict,
+                    'properties': _split_param(props)
+                })
+            else: # E.g. in case of lists (not yet implemented).
+                param_list.append({
+                    'name': name,
+                    'type': object,
+                    'unknown_props': props
+                })
+        return param_list
+
+    for status_code, respons_body in respons.items():
+        respons[status_code] = _split_param(respons_body)
+        if isinstance(status_code, int):
+            respons[str(status_code)] = respons.pop(status_code)
+
+    def _wrapper(func):
+        func.doc_info = {
+            'summary': descr,
```

Comment:
That's a good idea! But Ḯ'm wondering if this should be added as a 'description' instead, since (I guess?) docstrings may be longer than just one sentence? In the Open API Spec it is possible to describe each endpoint in two ways: 

- 'summary' : the oneliner that should be next to the POST /erasure_code_profile/{name} in the example.
- 'description': a longer description that can be displayed when you open up the endpoint and see more info.  

So far, only 'summary' is implemented. 

![summary and description](https://user-images.githubusercontent.com/22715268/52108979-2dafe800-25fc-11e9-934b-30e604448d11.jpg)
 'description'